### PR TITLE
[Enhencement](File Reader) delete old file_reader

### PR DESCRIPTION
**Pull request by BePPPower**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BePPPower** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.29 seconds
 stream load tsv:          474 seconds loaded 74807831229 Bytes, about 150 MB/s
 stream load json:         40 seconds loaded 2358488459 Bytes, about 56 MB/s
 stream load orc:          67 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230228161539_clickbench_pr_105920.html

File: be/src/exec/arrow/arrow_reader.cpp [hunk 1]
```diff
@@ -199,48 +199,42 @@ ArrowFile::~ArrowFile() {
...
     return ReadAt(_pos, nbytes, buffer);
 }
 
 arrow::Result<int64_t> ArrowFile::ReadAt(int64_t position, int64_t nbytes, void* out) {
-    int64_t reads = 0;
     int64_t bytes_read = 0;
     _pos = position;
-    while (nbytes > 0) {
-        Status result = _file->readat(_pos, nbytes, &reads, out);
+    int64_t total_reads = 0;
```
Comment (yiguolei) [inline be/src/exec/arrow/arrow_reader.cpp:216]: Any difference between bytes_read and total_reads?

File: be/src/util/broker_storage_backend.cpp [hunk 2]
```diff
@@ -72,14 +89,14 @@ Status BrokerStorageBackend::download(const std::string& remote, const std::stri
...
+    while (true) {
+        size_t read_len = 0;
+        Slice file_slice(reinterpret_cast<uint8_t*>(read_buf), buf_sz);
+        RETURN_IF_ERROR(broker_reader->read_at(cur_offset, file_slice, io_ctx, &read_len));
+        cur_offset += read_len;
+        if (read_len == 0) {
+            break;
         }
 
         if (read_len > 0) {
```
Comment (yiguolei) [inline be/src/util/broker_storage_backend.cpp:102]: This if is meaningless.

File: be/src/util/broker_storage_backend.cpp [hunk 3]
```diff
@@ -72,14 +89,14 @@ Status BrokerStorageBackend::download(const std::string& remote, const std::stri
     constexpr size_t buf_sz = 1024 * 1024;
     char read_buf[buf_sz];
```
Comment (yiguolei) [inline be/src/util/broker_storage_backend.cpp:90]: Do not use array to store buffer. Use a vector please. Array will allocate the buffer on stack and stack is very small.

File: be/src/util/hdfs_storage_backend.cpp [hunk 4]
```diff
@@ -183,13 +196,14 @@ Status HDFSStorageBackend::download(const std::string& remote, const std::string
     constexpr size_t buf_sz = 1024 * 1024;
     char read_buf[buf_sz];
```
Comment (yiguolei) [inline be/src/util/hdfs_storage_backend.cpp:197]: same problem here for read_buf.

File: be/src/util/hdfs_storage_backend.cpp [hunk 5]
```diff
@@ -206,19 +220,30 @@ Status HDFSStorageBackend::download(const std::string& remote, const std::string
...
+        cache_policy = io::FileCachePolicy::FILE_BLOCK_CACHE;
+    }
+    io::FileBlockCachePathPolicy file_block_cache;
+    io::FileReaderOptions reader_options(cache_policy, file_block_cache);
+    THdfsParams hdfs_params = parse_properties(_properties);
+    RETURN_IF_ERROR(FileFactory::create_hdfs_reader(hdfs_params, remote, &file_system, &hdfs_reader,
+                                                    reader_options, &io_ctx));
+
     constexpr size_t buf_sz = 1024 * 1024;
     char read_buf[buf_sz];
```
Comment (yiguolei) [inline be/src/util/hdfs_storage_backend.cpp:237]: same problem here.

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

**Pull request merged**